include/seqan/sequence/container_view_zipped.h

```diff
@@ -0,0 +1,266 @@
+//       notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//     * Neither the name of Knut Reinert or the FU Berlin nor the names of
+//       its contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ==========================================================================
+// Author: Rene Rahn <rene.rahn@fu-berlin.de>
+// ==========================================================================
+
+#ifndef INCLUDE_SEQAN_SEQUENCE_CONTAINER_VIEW_ZIPPED_H_
+#define INCLUDE_SEQAN_SEQUENCE_CONTAINER_VIEW_ZIPPED_H_
+
+namespace seqan
+{
+
+// ============================================================================
+// Forwards
+// ============================================================================
+
+// ============================================================================
+// Tags, Classes, Enums
+// ============================================================================
+
+// Tag to subclass basic container view.
+template <typename TSpec = void>
+struct ZipContainer;
+
+// Use tuple to store variadic container types.
+template <typename... TContPack, typename TSpec>
+class ContainerView<std::tuple<TContPack...>, ZipContainer<TSpec> >
+{
+public:
+    typedef typename Iterator<ContainerView, Standard>::Type TIterator;
+
+    TIterator   _begin;
+    TIterator   _end;
+
+    ContainerView() : _begin(), _end()
+    {
+        static_assert(sizeof...(TContPack) > 0, "Requries at least one argument!");
+    }
+
+    template <typename... TOtherContPack>
+    ContainerView(TOtherContPack && ...contArgs)
+    {
+        static_assert(sizeof...(TOtherContPack) > 0, "Requires at least one argument!");
+        static_assert(sizeof...(TOtherContPack) == sizeof...(TContPack), "Number of parameters don't match!");
+
+        // TODO: We need to assert if lenght of containers differ!
+
```

Comment:
No what I meant is that the length of all the containers that are zipped should have the same length (the same number of elements). The statement above just checks that the number of containers is the same.